packages/web3/src/index.ts

```diff
@@ -15,6 +15,258 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+/**
+ * This is the main (or 'umbrella') class of the web3.js library.
+ *
+ * ```ts
+ * import { Web3 } from 'web3';
+ *
+ * > Web3.utils //Not exported atm
+ * > Web3.version //Not exported atm
+ * > Web3.givenProvider
+ * > Web3.providers
+ * > Web3.modules //Not exported atm
+ * ```
+ *
+ * ### Web3.modules
+ * //Not implemented
+ *
+ * ### Web3 Instance
+ *
+ * The Web3 class is an umbrella package to house all Ethereum related modules.
+ *
+ * ```ts
+ * import { Web3 } from 'web3';
+ *
+ * // "Web3.givenProvider" will be set if in an Ethereum supported browser.
+ * const web3 = new Web3(Web3.givenProvider || 'ws://some.local-or-remote.node:8546');
+ *
+ * > web3.eth
+ * > web3.shh //not present
+ * > web3.bzz //not present
+ * > web3.utils //not exported atm
+ * > web3.version //not implemented
+ * ```
+ *
+ * ### version
+ *
+ * //Not implemented
+ * Contains the current package version of the web3.js library.
+ *
+ * ### utils
+ *
+ * Static accessible property of the Web3 class and property of the instance as well.
+ * //Not implemented
+ * ```ts
+ * Web3.utils
+ * web3.utils
+ * ```
+ *
+ * Utility functions are also exposed on the `Web3` class object diretly.
+ *
+ * ## setProvider
+ *
+ * ```ts
+ * web3.setProvider(myProvider)
+ * web3.eth.setProvider(myProvider)
+ * ...
+ * ```
+ *
+ * Will change the provider for its module.
+ *
+ * **_NOTE:_** When called on the umbrella package web3 it will also set the provider for all sub modules web3.eth  etc.
+ *
+ * See details: {@link Web3.setProvider}
+ *
+ * #### Example: Local Geth Node
+ * ```ts
+ * import { Web3 } from "web3";
+ * let web3: Web3 = new Web3('http://localhost:8545');
+ * // or
+ * let web3: Web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'));
+ *
+ * // change provider
+ * web3.setProvider('ws://localhost:8546');
+ * // or
+ * web3.setProvider(new Web3.providers.WebsocketProvider('ws://localhost:8546'));
+ *
+ * //todo add IPC provider
+ * ```
+ *
+ * @param provider A valid provider
```

Comment:
```suggestion
 *
 * @param provider A valid provider
 * #### Example: Local Geth Node
 * ```ts
 * import { Web3 } from "web3";
 * let web3: Web3 = new Web3('http://localhost:8545');
 * // or
 * let web3: Web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'));
 *
 * // change provider
 * web3.setProvider('ws://localhost:8546');
 * // or
 * web3.setProvider(new Web3.providers.WebsocketProvider('ws://localhost:8546'));
 *
 * //todo add IPC provider
 * ```
```
I think it would make more sense to put the parameters of `setProvider` before giving the example

Reply:
you are right, thanks